AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m

```diff
@@ -23,11 +36,171 @@ @interface MSDataStoreTests : XCTestCase
 @implementation MSDataStoreTests
 
 - (void)setUp {
-  [super setUp];
+    [super setUp];
 }
 
 - (void)tearDown {
-  [super tearDown];
+    [super tearDown];
 }
 
-@end
+- (void)testDefaultHeaderWithPartitionWithDictionaryNotNull {
+
+    // If
+    NSMutableDictionary *_Nullable additionalHeaders = [NSMutableDictionary new];
+    additionalHeaders[@"Type1"] = @"Value1";
+    additionalHeaders[@"Type2"] = @"Value2";
+    additionalHeaders[@"Type3"] = @"Value3";
+
+    // When
+    NSDictionary *dic = [MSCosmosDb defaultHeaderWithPartition:@"partition" dbToken:@"token" additionalHeaders:additionalHeaders];
+
+    // Then
+    XCTAssertNotNil(dic);
+    XCTAssertTrue(dic[@"Type1"]);
+    XCTAssertTrue(dic[@"Type2"]);
+    XCTAssertTrue(dic[@"Type3"]);
+}
+
+- (void)testDefaultHeaderWithPartitionWithDictionaryNull {
+
+    // If
+    NSDictionary *dic;
+
+    // When
+    dic = [MSCosmosDb defaultHeaderWithPartition:@"partition" dbToken:@"token" additionalHeaders:nil];
+
+    // Then
+    XCTAssertNotNil(dic);
+    XCTAssertTrue(dic[@"Content-Type"]);
+}
+
+- (void)testDocumentUrlWithTokenResultWithStringToken {
+
+    // If
+    MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithString:@"token"];
+
+    // When
+    NSString *result = [MSCosmosDb documentUrlWithTokenResult:tokenResult documentId:@"documentId"];
+
+    // Then
+    XCTAssertNotNil(result);
+}
+
+- (void)testDocumentUrlWithTokenResultWithObjectToken {
+
+    // If
+    MSTokenResult *tokenResult;
+    NSString *testResult;
+
+    // When
+    tokenResult = [[MSTokenResult alloc] initWithPartition:@"token"
+                                                 dbAccount:@"dbAccount"
+                                                    dbName:@"dbName"
+                                          dbCollectionName:@"dbCollectionName"
+                                                     token:@"token"
+                                                    status:@"status"
+                                                 expiresOn:@"expiresOn"];
+    testResult = [MSCosmosDb documentUrlWithTokenResult:tokenResult documentId:@"documentId"];
+
+    // Then
+    XCTAssertNotNil(testResult);
+    XCTAssertTrue([testResult isEqualToString:@"https://dbAccount.documents.azure.com/dbs/dbName/colls/dbCollectionName/docs/documentId"]);
+}
+
+- (void)testDocumentUrlWithTokenResultWithDictionaryToken {
+
+    // If
+    NSMutableDictionary *_Nullable tokenResultDictionary = [NSMutableDictionary new];
+    tokenResultDictionary[@"partition"] = @"partition";
```

Comment:
You set this token result dictionary up several times, it would be good to refactor it into a method

Reply:
Instead of taking parameters for each entry I'm thinking you could use constants for expected values